File: lib/limelight/builtin/players/radio_button.rb
#- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

on_cast do
  radio_button = Limelight::UI::Model::Inputs::RadioButtonPanel.new
  panel.add(radio_button)
  @radio_button = radio_button
end

attr_accessor :radio_button #:nodoc:
attr_reader :button_group #:nodoc:

# Sets the radio button group to which this radio button belongs.
#
def group=(group_name)
  @button_group = scene.button_groups[group_name]
  @button_group.add(@radio_button)
end

# Checks or unchecks this radio button.
#
def checked=(value)
  @radio_button.selected = value
end

alias :selected= :checked=

# Returns true is this radio is button is checked.
#
def checked
  return @radio_button.is_selected
end

alias :checked? :checked
alias :selected? :checked
alias :selected :checked

File: src/limelight/builtin/players/RadioButton.java
package limelight.builtin.players;

import limelight.ui.RadioButtonGroup;
import limelight.ui.events.panel.PanelEvent;
import limelight.ui.model.PropPanel;
import limelight.ui.model.inputs.RadioButtonPanel;

public class RadioButton
{

  private PropPanel propPanel;
  private RadioButtonPanel buttonPanel;

  public void install(PanelEvent event)
  {
    buttonPanel = new RadioButtonPanel();
    propPanel = (PropPanel)event.getRecipient();
    propPanel.add(buttonPanel);
    propPanel.getStagehands().put("radio-button", this);
  }

  public PropPanel getPropPanel()
  {
    return propPanel;
  }

  public RadioButtonPanel getButtonPanel()
  {
    return buttonPanel;
  }

  public boolean isSelected()
  {
    return buttonPanel.isSelected();
  }

  public void setSelected(boolean b)
  {
    buttonPanel.setSelected(b);
  }

  public void select()
  {
    buttonPanel.setSelected(true);
  }

  public String getGroup()
  {
    return buttonPanel.getGroup();
  }

  public void setGroup(String group)
  {
    buttonPanel.setGroup(group);
  }
}
